src/utils/process.rs

```diff
@@ -226,6 +253,13 @@ struct ProcInfo {
 }
 impl ProcInfo {
     fn new() -> Self {
+        // On Linux sysinfo optimizes for repeated process queries and keeps per-process
+        // /proc file descriptors (fds) open. This caching is not needed here, so
+        // set this to zero (this does nothing on other platforms).
+        // Also, there is currently a kernel bug which slows down syscalls when threads are
+        // involved (here: the ctrlc handler) and a lot of files are kept open.
```

Comment:
Out of interest, have you located this bug in the linux kernel issue tracker?

Reply:
So this is what allows us to fearlessly query many processes on linux again? Very interesting to see that you've got to the bottom of this problem.

Reply:
As the Linux bug tracker suggests, for now I have reported it to the Debian kernel team, see [#1001818](https://bugs.debian.org/1001818) for a minimal C and C++ example.

And yes, on Linux this makes process queries as fast as e.g. `ps auxf`.